### DISCUSSION TITLE
Use evcc with your Home Assistant sensor / tarif data

### DISCUSSION THREAD

**Discussion by HolgerMiara** [marq24/ha-evcc#50] [Q&A] [answered]
Moinauch!

Do you think it would be relatively easy to provide the electricity price data from Home Assistant to evcc? In Home Assistant, I use the integration [github.com/MTrab/energidataservice](https://github.com/MTrab/energidataservice), you can see the syntax of the data in the screenshot below. 

If complicated, I'll have to wait until it will be possible to calculate in evcc. ;-)

The issue is that we in Denmark have a dynamic (hourly) electricity price (which evcc can pull correctly). But on top, we pay time dependent network charges which have to be added. Currently evcc can either pull dynamic prices or deal with "tariff zones", but not a combination of both. 

![image](https://github.com/user-attachments/assets/84334d20-a59d-44e0-9f54-57cdb32c839a)

**Comment by marq24** [comment]
Good morning!

IMHO it would be much 'easier' to implement this in evcc directly (as PR) since we have source code that fetch the data correcly... So when the `today` list (build from the `raw_today` array is the final price, then it should be quite easy to adopt the existing evcc-tibber code to this.

But I could be totally wrong - since I did not fully picked up the meaning of 'But on top, we pay time dependent network charges which have to be added' - is this always a constant/fix/static delta that have to be added? [e.g. here for me in Germany I have this too - a fixed price addon of ~0,17€ per kWh that is more or less independant from the actual kWh stock price]

**Reply by marq24** [reply]
Ok - well - I found your evcc ticket - https://github.com/evcc-io/evcc/issues/17199 which explains the situation a bit more in detail - so the 'addon' is indeed time dependant - but the remaining question then (still) is, if the data you have provided in the screenshot `price` wise is the final (correct) overall price?

**Reply by marq24** [reply, marked as answer]
after spending some time with the evcc sources I think there is no easy/simple way to archive what you are looking for - IMHO in any case it's required to adjust/add code to evcc project in order to support your feature request [as you ahve already suggested @ evcc] - the only suggestion I can make (to enhance the request) is to make use of Month by "number" as list affected (instead of the "cryptic month-name-range' syntax...

Providing just "custom" and use a HA API URL as source will not work :-/

**Reply by HolgerMiara** [reply]
Moin Matthias, 

Thanks a million for giving it a thought! I thought so already, but simply had to ask... I had the slight hope that I maybe could create a sensor in HA which has a format that is understandable for evcc -- and then could be made available. 

However, I will keep waiting patiently for an implementation in evcc ;-) And as a workaround, it should still be possible to control low-cost charging of our cars from Home Assistant. 

Thanks again and have a nice Sunday!!
H.
